### PR TITLE
fix: enforce checking proofs when constraints are provided for a slot, solve TODOs

### PR DESCRIPTION
**Pull request by thedevbirb**
Now proofs are checked whenever the sidecar has accepted some constraints and any bid is rejected if it's doesn't contain them

### CODE REVIEW

**Review by merklefruit — APPROVED** [review]

**Comment by thedevbirb** [comment]
Tested on the devnet on works nicely on relay failures:
```
time="2024-11-05T18:28:12.004Z" level=warning msg="relay did not provide proofs despite existing constraints for slot" blockHash=0xfc915c704feaac8bdb3d2b7d1c272354cf4fbc06c35fb68af16eda23ba634c43 blockNumber=135 method=getHeaderWithProofs parentHash=0x0638799a5d6f6992c5cffde36cbd07362b26df4bd07600edd717ae6dad4cd5cc pubkey=0x8fda66b8607af873f4c2c8218dd3ffc7940d411047eb199b5cd010156af4845d21dd2e65b0e44cfffb5e78271e9bb29d slot=135 slotUID=180d64bf-bae8-434f-aae1-b712eecabb93 txRoot=0xdb885808af94f56baacbf6b6327b2661f76f3dd9a55c64db08efef7b9f91e970 ua=bolt-sidecar url="http://172.16.0.21:4040/eth/v1/builder/header_with_proofs/135/0x0638799a5d6f6992c5cffde36cbd07362b26df4bd07600edd717ae6dad4cd5cc/0x8fda66b8607af873f4c2c8218dd3ffc7940d411047eb199b5cd010156af4845d21dd2e65b0e44cfffb5e78271e9bb29d" value=0.000786569276767056 version=
```

Let's merge this once we're ready with a fix for the relay as well.
Once merged we should publish the new image with release tag `v0.3.0-alpha.rc3`.

**Comment by thedevbirb** [comment]
With a little more effort it also closes #9

File: server/constraints.go
Comment (merklefruit) [inline server/constraints.go]: can we return early instead of error?

File: server/service_test.go
Comment (merklefruit) [inline server/service_test.go]: CI is erroring here, can you double check?

File: server/service.go
Comment (merklefruit) [inline server/service.go:1339]: why are we skipping this now? Are we dropping Capella support altogether?

File: server/service.go
Comment (merklefruit) [inline server/service.go:280]: if you removed the AuctionTranscript type, it makes sense to also remove this

**Review by merklefruit — APPROVED** [review]
Great job on the early decoding

File: server/service.go
Comment (thedevbirb) [reply server/service.go:1339]: I started to remove some part of Capella support to address the TODO in #9. Is there any reason to keep Capella code now?

File: server/service.go
Comment (thedevbirb) [reply server/service.go:280]: This has been removed, what do you mean?

File: server/service.go
Comment (merklefruit) [reply server/service.go:1339]: I don't mind it, just wanted to understand why you did it

Comment (thedevbirb) [reply server/service.go:1339]: Just dropping unused code or useless (now) edge cases. We'll never accept Capella bids so we can safely remove everything related to it. The less code there is in this fork, the better!

File: server/service.go
Comment (merklefruit) [reply server/service.go:280]: Ah yeah my bad I thought the comment was kept, all good

**Comment by thedevbirb** [comment]
Finally ready and working in the devnet together with Helix: https://github.com/chainbound/helix/pull/21

File: server/proofs.go
Comment (merklefruit) [inline server/proofs.go]: Love the log

**Review by merklefruit — APPROVED** [review]
LGTM

File: server/proofs.go
Comment (thedevbirb) [reply server/proofs.go]: Oh wait, that should be removed actually ahah

**Pull request merged**
